test/webtest/src/test/java/com/onfido/qa/websdk/test/DocumentIT.java

```diff
@@ -0,0 +1,354 @@
+    }
+
+    @Test(description = "should be taken to the cross-device flow for video capture if there is no camera and docVideo variant requested")
+    public void testShouldBeTakenToTheCrossDeviceFlowForVideoCaptureIfThereIsNoCameraAndDocVideoVariantRequested() {
+        onfido().withSteps(new DocumentStep().withRequestedVariant(VIDEO))
+                .withDisableWebcam()
+                .init(IdDocumentSelector.class)
+                .select(PASSPORT, CrossDeviceIntro.class);
+    }
+
+    // @TODO: remove this test when we fully support docVideo variant for both desktop & mobile web
+    @Test(description = "should be taken to the cross-device flow for video capture docVideo variant requested")
+    public void testShouldBeTakenToTheCrossDeviceFlowForVideoCaptureDocVideoVariantRequested() {
+        onfido().withSteps(new DocumentStep().withRequestedVariant(VIDEO)).init(IdDocumentSelector.class).select(PASSPORT, CrossDeviceIntro.class);
+    }
+
+    @Test(description = "should be able to retry document upload when using customized API requests feature and receiving an error response from the callback")
+    public void testShouldBeAbleToRetryDocumentUploadWhenUsingCustomizedApiRequestsFeatureAndReceivingAnErrorResponseFromTheCallback() {
+
+        @SuppressWarnings("HardcodedLineSeparator") var promise = new Raw("() => Promise.reject({\n" + "    status: 422,\n" + "    response: JSON.stringify({\n" + "        error: {\n" + "            message: 'There was a validation error on this request',\n" + "            type: 'validation_error',\n" + "            fields: {detect_glare: ['glare found in image']}\n" + "        }\n" + "    })\n" + "})");
+
+        var enterpriseFeatures = new EnterpriseFeatures().withUseCustomizedApiRequests(true)
+                                                         .withOnSubmitDocument(promise)
+                                                         .withOnSubmitVideo(promise)
+                                                         .withOnSubmitDocument(promise);
+
+        var confirmUpload = onfido().withSteps("document", "complete")
+                                    .withEnterpriseFeatures(enterpriseFeatures)
+                                    .init(IdDocumentSelector.class)
+                                    .select(PASSPORT, DocumentUpload.class)
+                                    .clickUploadButton(ImageQualityGuide.class)
+                                    .upload(IDENTITY_CARD_WITH_GLARE)
+                                    .clickConfirmButton(ConfirmUpload.class);
+
+        verifyConfirmErrorMessage(confirmUpload, "glare");
+
+        confirmUpload.clickRedoButton(ImageQualityGuide.class)
+                     .upload(NATIONAL_IDENTITY_CARD_JPG)
+                     .clickConfirmButton(null);
+
+        // TODO: this test seem to have detected a real bug, the glare message stays, even if we uploaded a non glare image in the 2nd try
```

Comment:
Some images, specially if they are generated, tend to always return glare, but you should be able to proceed after the 2nd attempt, otherwise it's a bug

Reply:
Okay, I investigated into this and actually it's at least a bug of the test that I migrated. The test is using 

```
Onfido.init({
  "disableAnalytics" : true,
  "enterpriseFeatures" : {
    "useCustomizedApiRequests" : true,
    "onSubmitDocument" : () => Promise.reject({
    status: 422,
    response: JSON.stringify({
        error: {
            message: 'There was a validation error on this request',
            type: 'validation_error',
            fields: {detect_glare: ['glare found in image']}
        }
    })
}),
    "onSubmitVideo" : () => Promise.reject({
    status: 422,
    response: JSON.stringify({
        error: {
            message: 'There was a validation error on this request',
            type: 'validation_error',
            fields: {detect_glare: ['glare found in image']}
        }
    })
})
  },
  "language" : "en_US",
  "containerId" : "root",
  "steps" : [ "document", "complete" ],
  "token" : "..."
})
```

and handing over custom functions for submitting the document and video. It's always returning with an error, so a 2nd upload cannot be correct. We need to adjust the test. Please create a ticket for this. Thanks

Reply:
Good point. Created a ticket